x/ecocredit/genesis.go

```diff
@@ -60,6 +61,78 @@ func (s *GenesisState) Validate() error {
 		return err
 	}
 
+	// run each params validation method
+	if err := validateParams(s.Params); err != nil {
```

Comment:
How about `s.Params.Validate()` ?

Reply:
yeah that looks much cleaner -> https://github.com/regen-network/regen-ledger/pull/511/commits/9cd97542fa296479d1c1fe356240b09934819430